daliuge-translator/dlg/dropmake/lg.py

```diff
@@ -482,10 +487,26 @@ def _link_drops(self, slgn, tlgn, src_drop, tgt_drop, llink):
             tdrop.addStreamingInput(dropSpec_null, name="stream")
             self._drop_dict["new_added"].append(dropSpec_null)
         elif s_type in ["Application", "Control"]:
-            sname = slgn._getPortName("outputPorts")
+            sname = slgn._getPortName("outputPorts", index=-1)
             tname = tlgn._getPortName("inputPorts")
-            # logger.debug("Found port names: IN: %s, OUT: %s", sname, tname)
-            sdrop.addOutput(tdrop, name=sname)
+
```

Comment:
This code fixes an existing issue where we were not accurately keeping track of all output ports, and were instead only ever returning the final output port from the component's JSON (see edited code in lg_node.py).